Responsive, resizing, data normalization functions

djmj opened the pull request:
I added following features to use the signature plugin in a mobile responsive layout where you can switch from portrait to landscape view.

- **responsive** useful for for mobile switching between portrait and landscape
- **keep aspect ratio** responsive mode
- **normalize data to target dimension** if different applications and resolutions are used on the same data in database, the syncField can get normalized to a targeted dimension
- **allow resizing of signature if responsive**  on resize signature is not cleared but resized accordingly
- **init signature based on sync field** used for ajax responses to redraw it

As an attachment i have an test / demo site showing the functions and explanations.

@kbwood  I tried to send you a zip file by e-mail but the one on your website failed to send to

djmj committed:
Responsive, normalization, resize, load syncField

- **responsive** useful for for mobile switching between portrait and landscape
- **keep aspect ratio** responsive mode
- **normalize data to target dimension** for different applications and resolutions the data in syncField / database is normalized to a targeted dimension
- **allow resizing of signature if responsive** on resize signature is not cleared but resized accordingly
- **init signature based on sync field** useful for ajax responses to redraw it

djmj committed:
Demo for new responsive features

djmj committed:
use cdn resources

djmj committed:
Create core.js

djmj committed:
Create components-mobile.js

djmj committed:
Create jquery-plugins.js

djmj committed:
Rename test/resources/primefaces/jquery-plugins.js to test/resources/primefaces/jquery/jquery-plugins.js

djmj committed:
Create primefaces-mobile.css

djmj committed:
Update and rename demo/index.html to test/index.html

djmj committed:
Update README.md

djmj committed:
Update jquery.signature.js

Directly use javascript element instead of id lookup, since it was not working with special characters in id.

djmj committed:
Update README.md

djmj committed:
Update README.md

djmj committed:
Update jquery.signature.js

Remove resize window eventhandler in destroy using plugin event namespace to fix error on AJAX page updates.

djmj committed:
Add files via upload

djmj committed:
Delete jquery-mobile.css

djmj committed:
Add files via upload

wesslim commented:
Hi @djmj @kbwood  is this pull request still active?
I would like to try the **normalize data to target dimension**

djmj committed:
Update README.md

http://htmlpreview.github.io/?https://github.com/djmj/signature/blob/master/test/index.html Did not worked anymore due to CORS blocking resources. So added new URL.

djmj commented:
Hi @wesslim 

actually i only created a fork some years ago and did not want to create a merge. 
Most files are only used for my demo page in an primefaces environment.

If a merge should happen, only _jquery.signature.js_ should be merged and the other files ignored or moved to a demo folder. I never did a merge in github before so i am not sure what to do and i dont want to break @kbwood repository.

wesslim commented:
@djmj thanks for the explanation

djmj commented:
@wesslim 

You're welcome and yes its sadly there is a divergence in this fork and his latest update. Yesterday i updated the readme with a link to the demo page of my fork. So you can evaluate if it is usable. We are using this in our mobile hybrid-app for delivery receipts in production.

djmj committed:
Update README.md

djmj committed:
Update jquery.signature.js

Add option 'maxDimCompId' as a jquery selector that references a component (mostly a parent) that serves as a bounding box restricting the resizing to maximum width and height of this bounding box while keeping aspect ratio.
